analysisservice/src/test/java/nu/yona/server/analysis/service/AnalysisEngineServiceTests.java

```diff
@@ -460,58 +463,103 @@ public void activityOnNewDay()
+		Activity yesterdaysActivity = yesterdaysActivities.get(0);
+		assertThat("Expect right goal set to yesterday's activity", yesterdaysActivity.getActivityCategory(),
+				equalTo(gamblingGoal.getActivityCategory()));
+
+		// Verify one activity was created, with the right goal
+		List<Activity> activities = todaysDayActivity.getActivities();
+		assertThat("One activity created", activities.size(), equalTo(1));
+		Activity activity = activities.get(0);
+		assertThat("Expect right goal set to activity", activity.getActivityCategory(),
+				equalTo(gamblingGoal.getActivityCategory()));
+
+		assertThat("Expect new day", todaysDayActivity, not(equalTo(existingDayActivity)));
+		assertThat("Expect right date", todaysDayActivity.getStartDate(), equalTo(today.toLocalDate()));
+		assertThat("Expect activity added", todaysDayActivity.getLastActivity(), notNullValue());
+		assertThat("Expect matching start time", todaysDayActivity.getLastActivity().getStartTime(),
 				equalTo(startTime.toLocalDateTime()));
-		assertThat("Expect matching end time", newDayActivity.getValue().getLastActivity().getEndTime(),
+		assertThat("Expect matching end time", todaysDayActivity.getLastActivity().getEndTime(),
 				equalTo(endTime.toLocalDateTime()));
-		// Verify that a new activity was cached
-		verify(mockAnalysisEngineCacheService, times(1)).updateLastActivityForUser(eq(userAnonId), eq(gamblingGoal.getId()),
+
+		// Verify that there is an activity cached
+		verify(mockAnalysisEngineCacheService, atLeastOnce()).updateLastActivityForUser(eq(userAnonId), eq(gamblingGoal.getId()),
 				any());
+
 	}
 
 	@Test
 	public void appActivityCompletelyPrecedingLastCachedActivity()
 	{
 		ZonedDateTime now = ZonedDateTime.now(userAnonZoneId);
-		ZonedDateTime earlierActivityTime = now;
+		ZonedDateTime existingActivityTime = now;
 
-		DayActivity dayActivity = mockEarlierActivity(gamblingGoal, earlierActivityTime);
-		Activity earlierActivityEntity = dayActivity.getLastActivity();
+		DayActivity existingDayActivity = mockExistingActivity(gamblingGoal, existingActivityTime);
+		Activity existingActivity = existingDayActivity.getLastActivity();
 
 		ZonedDateTime startTime = now.minusMinutes(10);
 		ZonedDateTime endTime = now.minusSeconds(15);
 		service.analyze(userAnonId, createSingleAppActivity("Poker App", startTime, endTime));
 
 		// Verify that there is a new conflict message sent.
-		verify(mockMessageService, times(1)).sendMessageAndFlushToDatabase(any(), eq(anonMessageDestination));
+		verifyGoalConflictMessageSent(gamblingGoal);
+
 		// Verify that a database lookup was done finding the existing DayActivity to update
 		verify(mockDayActivityRepository, times(1)).findOne(userAnonId, now.toLocalDate(), gamblingGoal.getId());
+
 		// Verify that the day was updated
-		verify(mockDayActivityRepository, times(1)).save(dayActivity);
+		ArgumentCaptor<UserAnonymized> userAnonymizedCaptor = ArgumentCaptor.forClass(UserAnonymized.class);
```

Comment:
A method could be extracted for this day activity update pattern too. Then we can remove the comments :).